app/models/illiad_request.rb

```diff
@@ -4,54 +4,24 @@
 #  Class to handle creation of ILLiad OpenURL request
 ###
 class IlliadRequest
-  def initialize(scan)
-    @scan = scan
+  def initialize(request)
+    @request = request
   end
 
-  # rubocop:disable Metrics/MethodLength
-  def illiad_transaction_request
-    {
-      ProcessType: 'Borrowing',
-      RequestType: 'Article',
-      SpecIns: 'Scan and Deliver Request',
-      PhotoJournalTitle: scan_title,
-      PhotoArticleAuthor: @scan.authors,
-      Location: @scan.origin,
-      ReferenceNumber: @scan.origin_location,
-      PhotoArticleTitle: @scan.data[:section_title],
-      PhotoJournalInclusivePages: @scan.data[:page_range],
-      CallNumber: call_number,
-      ILLNumber: ill_number,
-      ItemNumber: item_number,
-      Username: @scan.user.sunetid
-    }.to_json
-  end
-  # rubocop:enable Metrics/MethodLength
-
   def request!
-    faraday_conn_w_req_headers.post('ILLiadWebPlatform/Transaction/', illiad_transaction_request)
+    faraday_conn_w_req_headers.post(
+      'ILLiadWebPlatform/Transaction/',
+      @request.illiad_request_params.merge(default_params).to_json
```

Comment:
```suggestion
      @request.illiad_request_params.reverse_merge(default_params).to_json
```

If they're default params, presumably the request-specific params should win?

Reply:
Thank you! I'll just merge the other direction; got the order wrong